solc/CommandLineParser.cpp

```diff
@@ -266,7 +266,14 @@ OptimiserSettings CommandLineOptions::optimiserSettings() const
 		settings.expectedExecutionsPerDeployment = optimizer.expectedExecutionsPerDeployment.value();
 
 	if (optimizer.yulSteps.has_value())
-		settings.yulOptimiserSteps = optimizer.yulSteps.value();
+	{
+		string const fullSequence = optimizer.yulSteps.value();
+		auto const delimiterPos = fullSequence.find(":");
+		settings.yulOptimiserSteps = fullSequence.substr(0, delimiterPos);
+
+		if (delimiterPos != string::npos)
+			settings.yulOptimiserCleanupSteps = fullSequence.substr(delimiterPos + 1);
```

Comment:
This also needs the assert I mentioned in https://github.com/ethereum/solidity/pull/13376/files#r950386578